src/index.ts

```diff
@@ -1001,3 +1001,65 @@ export class Values extends EventEmitter {
     return values;
   }
 }
+
+export type TokenState = {
+  token: PosNegToken;
+  clickSubmitted: boolean;
+};
+
+export const POS_NEG_STORAGE_KEY = "sajari_tokens";
+
+/**
+ * PosNegLocalStorageManager is a static utility class for manipulating Sajari's localstorage
+ * based management of PosNeg tokens. Typical use case is storing tokens for later consumption
+ * as users move through an ecommerce purcahse funnel.
+ */
+export class PosNegLocalStorageManager {
+  currentTokens: Record<string | number, TokenState>;
+  client: Client;
+  constructor(client: Client) {
+    const storageContent = localStorage.getItem(POS_NEG_STORAGE_KEY);
+    this.currentTokens = storageContent ? JSON.parse(storageContent) : {};
+    this.client = client;
+  }
+  add(fieldValue: string | number, token: PosNegToken) {
+    this.currentTokens[fieldValue] = { token, clickSubmitted: false };
+    localStorage.setItem(
+      POS_NEG_STORAGE_KEY,
+      JSON.stringify(this.currentTokens)
+    );
+  }
+  get(fieldValue: string | number) {
+    return this.currentTokens[fieldValue];
+  }
+  sendPosEvent(
+    fieldValue: string | number,
+    identifier: string,
+    weight: number
+  ) {
+    const tokenState = this.get(fieldValue);
+    if (tokenState.clickSubmitted !== false) {
```

Comment:
I think `tokenState` could be undefined?

Reply:
Shocked that TS didn't catch that. Thanks!